packages/dds/tree/src/simple-tree/api/verboseTree.ts

```diff
@@ -0,0 +1,409 @@
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} is a {@link VerboseTreeNode}.
+ * Any IFluidHandle values have been replaced by `THandle`.
+ * @privateRemarks
+ * This can store all possible simple trees, but it can not store all possible trees representable by our internal representations like FlexTree and JsonableTree.
+ */
+export type VerboseTree<THandle = IFluidHandle> =
+	| VerboseTreeNode<THandle>
+	| Exclude<TreeLeafValue, IFluidHandle>
+	| THandle;
+
+/**
+ * Verbose encoding of a {@link TreeNode}.
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} has an explicit `type` property, and `fields`.
+ * This allowed VerboseTreeNode to be unambiguous regarding which type each node is without relying on symbols or hidden state.
+ *
+ * Any IFluidHandle values have been replaced by `THandle`. If the `THandle` is JSON compatible, then this type is JSON compatible as well.
+ *
+ * @privateRemarks
+ * This type is only used for data which is copied into and out of the tree.
+ * When being copied out, its fine to have the data be mutable since its a copy.
+ *
+ * When being copied in, we don't need to mutate, so we could use a readonly variant of this type.
+ * however the copy in case (createFromVerbose) probably isn't harmed much by just reusing this type as is,
+ * since if the caller has immutable data, TypeScript doesn't prevent assigning immutable data to a mutable type anyway.
+ * Also relaxing the input methods to take readonly data would be a non-breaking change so it can be done later if desired.
+ *
+ * This format is simple-tree specialized alternative to {@link JsonableTree}.
+ * This format allows for all simple-tree compatible trees to be represented.
+ *
+ * Unlike `JsonableTree`, leaf nodes are not boxed into node objects, and instead have their schema inferred from the value.
+ * Additionally, sequence fields can only occur on a node that has a single sequence field (with the empty key) replicating the behavior of simple-tree ArrayNodes.
+ */
+export interface VerboseTreeNode<THandle = IFluidHandle> {
+	/**
+	 * The meaning of this node.
+	 * Provides contexts/semantics for this node and its content.
+	 * Typically used to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
+	 */
+	type: string;
+
+	/**
+	 * Content of this node.
+	 * For array nodes, an array of children.
+	 * For map and object nodes, an object which children under keys.
+	 * @remarks
+	 * For object nodes, the keys could be either the stored keys, or the property keys depending on usage.
+	 */
+	fields:
+		| VerboseTree<THandle>[]
+		| {
+				[key: string]: VerboseTree<THandle>;
+		  };
+}
+
+/**
+ * Options for how to interpret a `VerboseTree<TCustom>` when schema information is available.
+ */
+export interface ParseOptions<TCustom> {
+	/**
+	 * Fixup custom input formats.
+	 * @remarks
+	 * Main usage is translate some JSON compatible handle format into actual IFluidHandles.
```

Comment:
Nit ```suggestion
	 * Main usage is translate some JSON-compatible handle format into actual IFluidHandles.
```

Reply:
An example of this scenario would probably be good too. Also, currently the toJsonSchema logic throws if it encounters a Fluid Handle. If we have a reasonable representation we can support round-trip, we can probably lift that restriction.

Reply:
There is currently no way to encode a fluid handle using our public APIs. If there was using it would tend to result in GC issues as the data would disappear despite your custom reference to it existing.

Realistically there are 4 things you can currently do:
1. throw
2. leave handles as is (and not be json compatible)
3. Provide some domain specific encoding of handles (ex: I know this handle points to Foo, so I'll encode it as "container/foo")
4. use internal APIs to encode the handle somehow. One option is to make a non-round triable but at least comparable encoding (ex: pull out the handle absolute path or hash), the other would be to use the alpha+legacy fluid serializer and its round trippable format (the internal format, ISerializedHandle )

Regardless, you are right these docs need to be better.

Reply:
docs improved. Will likley need more before this goes public, but I think this is enough for our own internal test use in the short term.